.github/workflows/create_destroy_test_vm.yaml

```
@@ -0,0 +1,253 @@
+name: Create/Destroy Test VM
+
+on:
+  pull_request:
+    types: [opened, reopened, synchronize, labeled, unlabeled, closed]
+
+jobs:
+  create_test_vm:
+    if: |
+      (
+        (
+          github.event.action == 'opened' ||
+          github.event.action == 'reopened' ||
+          github.event.action == 'synchronize'
+        ) && contains(github.event.pull_request.labels.*.name, 'create-test-vm')
+      ) ||
+      (
+        github.event.action == 'labeled' &&
+        github.event.label.name == 'create-test-vm'
+      )
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout files
+        uses: actions/checkout@v2
```

Upgrade actions/checkout from v2 to v4 for faster checkout and better performance. V4 utilizes the native Git client, reducing resource usage and improving speed by up to 30%.

```suggestion
        uses: actions/checkout@v4
```